client/TrailSeek/src/app/userSlice.js

```diff
@@ -67,6 +72,27 @@ export const signUp = createAsyncThunk(
   }
 );
 
+export const fetchUserData = createAsyncThunk(
+  "user/fetchUserData",
+  async (_, { rejectWithValue, getState }) => {
+    try {
+      const response = await trailSeek.get("/user", {
+        headers: { Authorization: `Bearer ${getState().user.profile.token}` },
```

Comment:
Is it possible to move this to the trailSeek API in /src/api/trailSeek.js so we don't have to always set the authorisation token manually?

Reply:
Right now that file is used for all the trails and other forms of requests. I can probably do it. Will look into it.

Reply:
The idea is to have all the headers set automatically whenever a request is made to the trailseek APIs. Maybe one way to do it is instantiate the API with the state [getState()]. Just an idea, not sure if that would work for all cases.